docs/getting-started/getting-started-with-docker.rst

```diff
@@ -22,15 +24,143 @@ There is a good chance that you have heard of containerization. A container is e
 
 Users of HPC (High Performance Compute) systems may not be able to run Docker Engine directly, depending on your sysadmin's policies and the details of your HPC's scheduling system. We can't go over all of possible HPC setups here, but we recommend checking out `Singularity <https://sylabs.io/guides/2.6/user-guide/singularity_and_docker.html>`__ and `Shifter <https://github.com/NERSC/shifter>`__, both of which are designed for running Docker images without using the Docker Engine program. If your HPC does not support Docker Engine, there is a strong chance it is already set up with an alternative.
 
-For the sake of simplicity, this tutorial will assume you are running Docker Engine on a non-HPC Unix-like system.
+For the sake of simplicity, this tutorial will assume you are running Docker on a non-HPC Unix or Unix-like system.
+
+Thick vs thin Docker images
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Docker images are sometimes described in terms of thick versus thin/slim, describing how much stuff is inside them. These are relative terms, not a precise definition, but they can be useful shorthands for two different schools of thought around Docker: Including a single tool, or multiple tools.
+
+Over time, we at Dockstore have found that slim Docker images, those with single tools installed in them, are more helpful for extending and building new workflows with. That being said, thick Docker containers, which include multiple tools and even full workflows with frameworks like `SeqWare <https://seqware.github.io/>`__ or `Galaxy <https://galaxyproject.org/>`__, can have their place as well. Projects like the ICGC `PanCancer Analysis of Whole Genomes <https://dcc.icgc.org/pcawg>`__ (PCAWG) made use of thick Docker containers that had complex workflows that fully encapsulated alignment and variant calling. Another example is the `UWGAC Analysis Workflows <https://dockstore.org/organizations/bdcatalyst/collections/UWGACAncestryRelatedness>`__, which all use the same Docker image full of R scripts in order to preform everything from file conversion to a full GWAS analysis. The self-contained nature of these Docker containers allow for mobility between a wide variety of environments, and sometimes simplify the setup of these pipelines across a wide variety of HPC and cloud environments. Either approach works for the Dockstore so long as you can describe the tool or workflow inside the Docker container as a CWL/WDL/NFL-defined tool (which you can for most things).
+
+Using existing Docker images
+----------------------------
+Sometimes, the tool you want to run is already Dockerized. Perhaps you want to use that existing image for your own research, or maybe you want to base a pipeline off it. But how would you know that such an image exists? Where do you look for Docker images? And how do you know if you've found a good one?
```

Comment:
Point out Biocontainers as a resource here?

Reply:
I am slightly hesitant because of https://github.com/dockstore/dockstore/issues/4945 -- specifically the part about the fact there's a biocontainers version of bamstats that may or may not be outdated, and might not even be by Dockstore. bamstats is used in the tutorial, so someone following our own advice may end up with an unknown image.

Definitely willing to give biocontainers a shoutout eventually, perhaps the issue is that I lack understanding in how they handle versioning -- but for now I think we should wait for https://github.com/dockstore/dockstore/issues/4945 to be handled first.